packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-group.tsx

```diff
@@ -1,57 +1,102 @@
-import { Component, h, JSX, Element, Host, Listen } from '@stencil/core';
+import { Component, h, JSX, Element, Host, Listen, Prop } from '@stencil/core';
 import { trackComponent } from '@utils/tracking/usage';
 import { logWarn } from '@utils/error/log-error';
 import {
   setFocusTarget,
-  resetFocusableSibling
+  resetFocusableSibling,
+  GROUP_QUANTITIES
 } from './gux-avatar-group.service';
+import { GuxAvatarGroupQuantity } from './gux-avatar-group.types';
+import { GuxAvatarAccent } from '../gux-avatar/gux-avatar.types';
+
+interface ProcessedGroupItem {
+  img?: HTMLImageElement | null;
+  name: string;
+  accent: GuxAvatarAccent;
+  isOverflow: boolean;
+  groupItem: HTMLGuxAvatarGroupItemBetaElement;
+}
 
 /**
  * @slot - slot for gux-avatar-group-item components
  */
 @Component({
   styleUrl: 'gux-avatar-group.scss',
   tag: 'gux-avatar-group-beta',
-  shadow: { delegatesFocus: true }
+  shadow: true
 })
 export class GuxAvatarGroup {
+  private processedGroupItems: ProcessedGroupItem[] = [];
+
   @Element()
   root: HTMLElement;
 
+  @Prop()
+  quantity: GuxAvatarGroupQuantity = Math.max(
```

Comment:
I think this is missing the type declaration? I see we set the type of the default value, but not the variable itself iirc. Meaning that the number isn't limited, which leads to my next questions. Are we limiting the quantity to a max of 7 to start and that's why we have the number array stuff?